src/ORM/DataObject.php

```diff
@@ -3724,6 +3724,32 @@ public function onAfterBuild()
         $this->extend('onAfterBuild');
     }
 
+    /**
+     * $fieldPath List of paths on this object.
+     * @return mixed Returns DBField of the field on the object or null if there is not relations between fields.
+     */
+    private function hasRelations(string $fieldPath)
+    {
+        $component = $this;
+        foreach (explode('.', $fieldPath ?? '') as $relation) {
+            if (!$component) {
+                return false;
+            }
+
+            if (ClassInfo::hasMethod($component, $relation)) {
+                if ($component->hasField($relation)
+                    || (($component instanceof DataObject) && $component->getRelationType($relation) !== null)) {
+                    $component = $component->$relation();
+                } else {
+                    return false;
```

Comment:
Any `$relation` that does not pass this check is not suitable for further passing to the `search` method.